### PR TITLE
feat: add AssumeRolePolicyDocument to AWS::Serverless::Function

### PR DESCRIPTION
**Pull request by timoschilling**
*Issue #, if available:*
To use a `AWS::Serverless::Function` as lambda@edge you need to add an explicit policy to allow CloudFront to execute the function.

*Description of changes:*
With this change you can define a `AssumeRolePolicyDocument` on the SAM created role.

Before:
```yaml
  Function:
    Type: AWS::Serverless::Function
    Properties:
      Handler: index.handler
      Runtime: nodejs8.10
      Timeout: 30
      AutoPublishAlias: true
      Role: !GetAtt Role.Arn
      InlineCode: !Sub |
        ...

  Role:
    Type: AWS::IAM::Role
    Properties:
      ManagedPolicyArns:
        - arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
      AssumeRolePolicyDocument:
        Version: 2012-10-17
        Statement:
          - Effect: Allow
            Principal:
              Service:
                - lambda.amazonaws.com
                - edgelambda.amazonaws.com
            Action: sts:AssumeRole
      Path: /
      Policies:
        - PolicyName: InvokeLambda
          PolicyDocument:
            Version: 2012-10-17
            Statement:
              - Effect: Allow
                Action: s3:*
                Resource: ...
```

After:
```yaml
  Function:
    Type: AWS::Serverless::Function
    Properties:
      Handler: index.handler
      Runtime: nodejs8.10
      Timeout: 30
      AutoPublishAlias: true
      Policy:
        Version: 2012-10-17
        Statement:
          - Effect: Allow
            Action: s3:*
            Resource: ...
      AssumeRolePolicyDocument:
        Version: 2012-10-17
        Statement:
          - Effect: Allow
            Principal:
              Service:
                - lambda.amazonaws.com
                - edgelambda.amazonaws.com
            Action: sts:AssumeRole
      InlineCode: !Sub |
        ...
```

*Description of how you validated changes:*

*Checklist:*

- [x] Write/update tests
- [x] `make pr` passes
- [x] Update documentation
- [x] Verify transformed template deploys and application functions as expected
- [x] Add/update example to `examples/2016-10-31`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: examples/2016-10-31/lambda_edge/template.yaml [hunk 1]
```diff
@@ -61,24 +61,14 @@ Resources:
...
-          AutoPublishAlias: live 
-
-    LambdaEdgeFunctionRole:
-      Type: "AWS::IAM::Role"
-      Properties:
-          Path: "/"
-          ManagedPolicyArns:
-              - "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
+          AutoPublishAlias: live
           AssumeRolePolicyDocument:
```
Comment (praneetap) [inline examples/2016-10-31/lambda_edge/template.yaml]: @timoschilling Thanks for the contribution! Is there a reason why this is not supported in globals? To make that change you would need to add it [here](https://github.com/awslabs/serverless-application-model/blob/develop/samtranslator/plugins/globals/globals.py#L40)

Comment (timoschilling) [reply examples/2016-10-31/lambda_edge/template.yaml]: I never think about that. I will add it to globals

Comment (timoschilling) [reply examples/2016-10-31/lambda_edge/template.yaml]: done

**Comment by timoschilling** [comment]
@praneetap @ShreyaGangishetty any updates on this PR?

**Comment by ShreyaGangishetty** [comment]
@timoschilling Thank you for your contribution! This PR looks good to me

**Comment by jplock** [comment]
With this change, does the Lambda@Edge function still have access to write to CloudWatch logs, which previously was part of `arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole`?

**Comment by jplock** [comment]
Is `AssumeRolePolicyDocument` supported in `Globals`?

**Pull request merged**